File: src/main/java/com/dsm/hwanghae/domains/service/DiaryBookService.java
package com.dsm.hwanghae.domains.service;

import com.dsm.hwanghae.utils.form.DiaryBookResponseForm;
import com.google.gson.GsonBuilder;
import okhttp3.OkHttpClient;
import org.springframework.stereotype.Service;
import retrofit2.Response;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;
import retrofit2.converter.scalars.ScalarsConverterFactory;

import java.io.IOException;

@Service
public class DiaryBookService {

    public DiaryBookResponseForm getDiaryBooks(String userId) {

        Retrofit retrofit = new Retrofit.Builder()
                .baseUrl("http://gyeongsang:8893")
                .client(new OkHttpClient().newBuilder().build())
                .addConverterFactory(ScalarsConverterFactory.create())
                .addConverterFactory(GsonConverterFactory.create(new GsonBuilder().setLenient().create()))
                .build();

        DiaryBookRequestConnectionService service = retrofit.create(DiaryBookRequestConnectionService.class);

        Response<DiaryBookResponseForm> response = null;
        try {
            response = service.addRequest(userId).execute();
            System.out.println("name : " + response.body().getDiaryBooks().get(0).getName());
        } catch(IOException e) {
            e.printStackTrace();
        }

        return response.body();
    }
}

File: src/main/java/com/dsm/hwanghae/domains/service/DiaryBookService.kt
package com.dsm.hwanghae.domains.service

import com.dsm.hwanghae.utils.form.DiaryBookResponseForm
import org.springframework.stereotype.Service
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

@Service
class DiaryBookService {

    fun getDiaryBooks(id: String): DiaryBookResponseForm {

        val retrofit = Retrofit.Builder()
                .baseUrl("http://localhost:8888")
                .addConverterFactory(GsonConverterFactory.create())
                .build()

        val service = retrofit.create(DiaryBookRequestConnectionService::class.java)

        var diaryBooks = DiaryBookResponseForm(ArrayList())
        service.addRequest(id)?.enqueue(object: Callback<DiaryBookResponseForm> {
            override fun onFailure(call: Call<DiaryBookResponseForm>, t: Throwable) {
                TODO("실패했을 경우 어떻게 할 거냐")
                print("실패함");
            }

            override fun onResponse(call: Call<DiaryBookResponseForm>, response: Response<DiaryBookResponseForm>) {
                diaryBooks = response.body()!!
            }
        })

        return diaryBooks
    }
}